### PR TITLE
Update certificate workflow for istio 1.9

### PR DESCRIPTION
**Pull request by joekelley**

### CODE REVIEW

File: pkg/certificates/agent/reconciliation/cert_agent_reconciler.go [hunk 1]
```diff
@@ -280,9 +282,11 @@ func (r *certAgentReconciler) reconcileIssuedCertificate(
 }
 
 // bounce (delete) the listed pods
-// returns true if we need to wait for replacement pods before proceeding to process the podBounceDirective.
+// returns true if we need to wait before proceeding to process the podBounceDirective.
+// we might wait if we need the istiod control plane to come back online or if we need to ensure
+// that the root cert has been propagated to all namespaces for consumption by the data plane.
```
Comment (harveyxia) [inline pkg/certificates/agent/reconciliation/cert_agent_reconciler.go:287]:
```suggestion
// we must wait for the following conditions: 
// 1. istiod control plane has come back online after it has been restarted
// 2. istio's root cert has been propagated to all istio-controlled namespaces for consumption by the data plane.
```

File: pkg/certificates/agent/reconciliation/cert_agent_reconciler.go [hunk 2]
```diff
@@ -259,7 +261,7 @@ func (r *certAgentReconciler) reconcileIssuedCertificate(
 			}
 
 			// try to bounce the pods and see if we need to wait
-			waitingForReplacements, err := r.bouncePods(podBounceDirective, inputPods)
+			waitingForReplacements, err := r.bouncePods(podBounceDirective, inputPods, inputConfigMaps, inputSecrets)
```
Comment (harveyxia) [inline pkg/certificates/agent/reconciliation/cert_agent_reconciler.go:264]:
```suggestion
			waitForConditions, err := r.bouncePods(podBounceDirective, inputPods, inputConfigMaps, inputSecrets)
```

File: pkg/certificates/agent/reconciliation/cert_agent_reconciler.go [hunk 3]
```diff
@@ -314,6 +320,31 @@ func (r *certAgentReconciler) bouncePods(podBounceDirective *v1alpha2.PodBounceD
 			continue
 		}
 
+		if selector.RootCertSync != nil {
+			configMap, err := allConfigMaps.Find(selector.RootCertSync.ConfigMapRef)
+			if err != nil && errors.IsNotFound(err) {
+				// ConfigMap isn't found; let's wait for it to be added by Istio
+				time.Sleep(time.Second)
```
Comment (ilackarms) [inline pkg/certificates/agent/reconciliation/cert_agent_reconciler.go:327]: would be good to put a logline here i think

File: pkg/certificates/agent/reconciliation/cert_agent_reconciler.go [hunk 4]
```diff
@@ -314,6 +320,31 @@ func (r *certAgentReconciler) bouncePods(podBounceDirective *v1alpha2.PodBounceD
...
+
+			secret, err := allSecrets.Find(selector.RootCertSync.SecretRef)
+			if err != nil {
+				return true, err
+			}
+
+			if configMap.Data[selector.RootCertSync.ConfigMapKey] != string(secret.Data[selector.RootCertSync.SecretKey]) {
+				// the configmap's public key doesn't match the root cert CA's
+				// sleep to allow time for the cert to be distributed and retry
+				time.Sleep(time.Second)
```
Comment (ilackarms) [inline pkg/certificates/agent/reconciliation/cert_agent_reconciler.go:342]: here as well

**Review by ilackarms — COMMENTED** [review]
lgtm, 2 minor comments

**Review by ilackarms — APPROVED** [review]

**Pull request merged**